iml-wire-types/src/warp_drive.rs

```diff
@@ -0,0 +1,394 @@
+use crate::{
+    db::{
+        Id, LnetConfigurationRecord, ManagedTargetMountRecord, OstPoolOstsRecord, OstPoolRecord,
+        StratagemConfiguration, VolumeNodeRecord,
+    },
+    Alert, Filesystem, Host, LockChange, Target, TargetConfParam, Volume,
+};
+use im::{HashMap, HashSet};
+use std::{
+    hash::{BuildHasher, Hash},
+    iter::FusedIterator,
+    ops::Deref,
+    sync::Arc,
+};
+
+/// This trait is to bring method `arc_values()` to the collections of
+/// type HashMap<K, Arc<V>> to simplify iterating through the values.
+/// Example:
+/// ```rust,norun
+///     use std::sync::Arc;
+///     use im::HashMap;
+///     use iml_wire_types::warp_drive::ArcValuesExt;
+///
+///     let hm: HashMap<i32, Arc<String>> = im::hashmap!(
+///         1 => Arc::new("one".to_string()),
+///         2 => Arc::new("two".to_string()),
+///     );
+///     let vals: Vec<&str> = hm.arc_values().map(|s| &s[..]).collect();
+///     println!("values are: {:?}", vals);
+/// ```
+pub trait ArcValuesExt<K, V> {
```

Comment:
Did [Deref behavior](https://doc.rust-lang.org/std/sync/struct.Arc.html#deref-behavior) not work out well?